### PR TITLE
Implemented Enter key to add taxonomy term

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
Hi
In this PR I added a feature , Now user can press enter/return key to add term to vocabulary.

https://github.com/mitodl/lore/issues/272

@pdpinch @pwilkins @carsongee 

screen shot 2015-07-06 at 5 14 37 pm

### CODE REVIEW

**Comment by pwilkins** [comment]
@amir-qayyum-khan Please change the commit message to, "Implemented Enter key to add taxonomy term."  The commit messages become our release notes, so we want the commit messages to follow a consistent pattern.  All should be in past tense.

**Comment by noisecapella** [comment]
You should also squash your commits into one commit once it's ready for review, and use the 'Needs Review' tag

**Comment by amir-qayyum-khan** [comment]
Done @noisecapella 

cc @pwilkins

**Comment by noisecapella** [comment]
Next time please add the 'Needs Review' tag when your PR is ready for review.

I ran this on my computer and it looks fine to me. However tests are failing. Can you identify and squash a new commit which fixes the tests?

**Comment by noisecapella** [comment]
Sorry, @carsongee informed me that you can't edit the label, feel free to make a comment here instead to let me know

**Comment by amir-qayyum-khan** [comment]
Fixed issue @noisecapella , I dont have access to add label.

**Comment by noisecapella** [comment]
Can you add a javascript test for this? You can add a file in `ui/jstests` and use `test-facets.js` as an example.

Some information about testing React: https://facebook.github.io/react/docs/test-utils.html

**Comment by noisecapella** [comment]
This would require substantial changes to the `manage_taxonomies.jsx` file in order to be able to test those React components, but it would be worth seeing an attempt at React testing since we haven't done this before in this project.

**Comment by amir-qayyum-khan** [comment]
@carsongee @noisecapella please review

cc @pwilkins

File: ui/static/ui/js/listing.js [hunk 1]
```diff
@@ -224,6 +224,7 @@ define(
       });
 
       var repoSlug = $("#repo_slug").val();
-      setupManageTaxonomies(repoSlug);
+      console.log(repoSlug);
```
Comment (carsongee) [inline ui/static/ui/js/listing.js]: you should remove this

**Comment by carsongee** [comment]
Other than that, looks great :bouquet: :tangerine:

File: ui/jstests/test_manage_taxonomies.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      var randomEvent = $.Event("keyup", {keyCode: 64});
+      var eventEnter = $.Event("keyup", {keyCode: 13});
+
+      assert.equal(
+        false, vocabularyComponentRendered.onEnterPress(randomEvent)
+      );
+      assert.equal(
+        true, vocabularyComponentRendered.onEnterPress(eventEnter)
+      );
+
```
Comment (carsongee) [inline ui/jstests/test_manage_taxonomies.jsx]: Actually, can you add an assertion here that the state has been updated with the item?

File: ui/static/ui/js/require_config.js [hunk 3]
```diff
@@ -11,7 +11,8 @@ var REQUIRE_PATHS = {
   csrf: "../ui/js/csrf",
   facets: "../ui/js/facets",
   listing: "../ui/js/listing",
-  setup_manage_taxonomies: "../ui/js/manage_taxonomies.jsx?noext"
+  setup_manage_taxonomies: "../ui/js/manage_taxonomies.jsx?noext",
+  jquery_mockjax: "../node_modules/jquery-mockjax/src/jquery.mockjax"
```
Comment (noisecapella) [inline ui/static/ui/js/require_config.js]: This line should be removed, the path is already added to `test-main.js` which sets up paths specific to the tests. We shouldn't ever need the jQuery mocking library outside of our tests.

**Comment by amir-qayyum-khan** [comment]
Fix @carsongee and @noisecapella 

cc @pwilkins

File: ui/static/ui/js/manage_taxonomies.jsx [hunk 4]
```diff
@@ -49,8 +50,18 @@ define('setup_manage_taxonomies', ['reactaddons', 'lodash', 'jquery'],
         </li>
       </ul>;
     },
+    onEnterPress: function(e) {
+      var code = (e.keyCode ? e.keyCode : e.which);
+      if (code === 13) { //Enter keycode
+        this.handleAddTermClick();
+        return true;
```
Comment (noisecapella) [inline ui/static/ui/js/manage_taxonomies.jsx]: React does not support returning boolean values in event handlers, you need to use `event.preventDefault()` instead. I just learned about this yesterday, see [first warning box](https://facebook.github.io/react/docs/events.html)

File: ui/jstests/test_manage_taxonomies.jsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+            vocabulary={vocabulary}
+            terms={vocabulary.terms}
+            reportError={reportError}
+            repoSlug="repo"
+          />
+        );
+      var randomEvent = $.Event("keyup", {keyCode: 64});
+      var eventEnter = $.Event("keyup", {keyCode: 13});
+
+      assert.equal(
```
Comment (noisecapella) [inline ui/jstests/test_manage_taxonomies.jsx]: You will need to remove these two asserts due to the lack of return but you should be able to just add an assert for the number of terms in `vocabularyComponentRendered.state.terms` to confirm that nothing changed

Comment (amir-qayyum-khan) [inline ui/jstests/test_manage_taxonomies.jsx]: Fix @noisecapella

**Comment by noisecapella** [comment]
Everything looks great. It's nice to have `manage_taxonomies.jsx` fully covered! I'll merge when tests pass

**Comment by amir-qayyum-khan** [comment]
@noisecapella build is ready

cc @carsongee @pwilkins

**Pull request merged**